packages/account/src/providers/provider.ts

```diff
@@ -460,8 +483,17 @@ export default class Provider {
    * @returns A promise that resolves to a Provider instance.
    */
   static async create(url: string, options: ProviderOptions = {}): Promise<Provider> {
-    const provider = new Provider(url, options);
+    const [urlToUse, addBasicAuth] = this.handleBasicAuth(url);
+    const provider = new Provider(urlToUse, {
+      ...options,
+      requestMiddleware: async (request) => {
+        await addBasicAuth?.(request);
+        return options.requestMiddleware?.(request) ?? request;
+      },
```

Comment:
Injecting auth conditionally within the flow.

```suggestion
      requestMiddleware: async (request: <TypeHere>) => {
        if (auth) {
          request.headers ??= {};
          request.headers.Authorization = auth;
        }

        return options.requestMiddleware?.(request) ?? request;
      },
```